crates/ra_ide_api/src/diagnostics.rs

```diff
@@ -46,6 +49,39 @@ pub(crate) fn diagnostics(db: &RootDatabase, file_id: FileId) -> Vec<Diagnostic>
             severity: Severity::Error,
             fix: Some(fix),
         })
+    })
+    .on::<hir::diagnostics::MissingFields, _>(|d| {
+        let file_id = d.file().original_file(db);
+        let source_file = db.parse(file_id);
+        let mut buff = String::new();
+        let syntax_node = d.syntax_node_ptr();
+        let node = syntax_node.to_node(&source_file);
+        let insert_to = if let Some(last_child) = node.last_child() {
+            buff.push_str(", ");
+            last_child.range().end()
+        } else {
+            let node_range = syntax_node.range();
+            if node_range.len().to_usize() == 2 {
+                buff.push(' ');
+            }
+            node_range.end() - TextUnit::of_char('}')
+        };
+        for (i, f) in d.missed_fields.iter().enumerate() {
+            write!(buff, "{}: ()", f).unwrap();
+            if i != d.missed_fields.len() - 1 {
+                buff.push_str(", ");
+            }
+        }
+        let mut builder = TextEditBuilder::default();
+        builder.insert(insert_to, buff);
+        let fix =
+            SourceChange::source_file_edit_from("fill struct fields", file_id, builder.finish());
```

Comment:
In https://github.com/rust-analyzer/rust-analyzer/pull/1184, we've added a structured editing API, which should be more pleasant to use, and, hopefully, more correct, than adding fields by hand.